cocos2d/core/components/editbox/WebEditBoxImpl.js

```diff
@@ -434,6 +434,7 @@ Object.assign(WebEditBoxImpl.prototype, {
         } else if(inputMode === InputMode.PHONE_NUMBER) {
             type = 'number';
             elem.pattern = '[0-9]*';
+            elem.onmousewheel = 'return false;';
```

Comment:
```suggestion
            elem.onmousewheel = function () { return false; };
```

Reply:
Thanks for feedback
we thought that the eval string is not a good practice, and it doesn't work well on the Chrome on MacOS

maybe it is better to change like this.

Reply:
I have no local compilation environment, so I directly get the input object from the implementation of Cocos Creator 2.4.2:
I'll create a new Editbox on a blank scene and test it as follows:
```
   onLoad: function () {
        this.editBox._impl._elem.setAttribute("onmousewheel","function () { return false; };");
        console.log( this.editBox._impl._elem);
    },
```
chrome report error:
![image](https://user-images.githubusercontent.com/34433563/128299056-c8a7c677-5840-4867-9b04-d865dc99dbb6.png)


Using native HTML5 tests, error as above (missing function name):
```
 <form>
        name: <input type="number" name="number" onmousewheel="function a(){ return false; };"><br>
</form>
```
After add a function name：
```
<form>
        name: <input type="number" name="number" onmousewheel="function a(){ return false; };"><br>
 </form>

 this.editBox._impl._elem.setAttribute("onmousewheel","function a() { return false; };");
```
The mouse wheel effect does not fail

Reply:
## the difference is that
```
elem.onmousewheel = "function () { return false; };"
```
means we **declare** a nameless function at the mousewheel callback


```
elem.onmousewheel = function () { return false; };
```
means we **call** the namelesss function  at the mousewheel callback

Reply:
Thank you for your answer. I learned a lot. I am a back-end developer，I just started working on cocos. 😸